pyomo/contrib/solver/solvers/ipopt.py

```diff
@@ -499,46 +535,136 @@ def solve(self, model, **kwds):
+            parsed_data['iters'] = int(iter_match.group(1))
+        # Gather all the iteration data
+        iter_table = re.findall(r'^(?:\s*\d+.*?)$', output, re.MULTILINE)
+        if iter_table:
+            columns = [
+                "iter",
+                "objective",
+                "inf_pr",
+                "inf_du",
+                "lg_mu",
+                "d_norm",
+                "lg_rg",
+                "alpha_du",
+                "alpha_pr",
+                "ls",
+            ]
+            all_iterations = []
+
+            for line in iter_table:
+                tokens = line.strip().split()
+                if len(tokens) == len(columns):
+                    iter_data = dict(zip(columns, tokens))
+
+                    # Extract restoration flag from 'iter'
+                    iter_val = iter_data["iter"]
+                    iter_match = re.match(r"(\d+)(r?)", iter_val)
+                    if iter_match:
+                        iter_data["restoration"] = bool(
+                            iter_match.group(2)
+                        )  # True if 'r' is present
+                    else:
+                        iter_data["restoration"] = False
+                    iter_data.pop('iter')
+
+                    # Separate alpha_pr into numeric part and optional tag
+                    alpha_pr_val = iter_data["alpha_pr"]
+                    match = re.match(r"([0-9.eE+-]+)([a-zA-Z]?)", alpha_pr_val)
+                    if match:
+                        iter_data["alpha_pr"] = match.group(1)
+                        iter_data["step_acceptance"] = (
+                            match.group(2) if match.group(2) else None
+                        )
+                    else:
+                        iter_data["step_acceptance"] = None
+
+                    # Attempt to cast all values to float where possible
+                    for key in iter_data:
+                        try:
+                            if iter_data[key] == '-':
+                                iter_data[key] = None
+                                continue
+                            if isinstance(iter_data[key], bool):
+                                continue
+                            iter_data[key] = float(iter_data[key])
+                        except (ValueError, TypeError):
+                            pass
+
+                    all_iterations.append(iter_data)
+
+            parsed_data['iteration_log'] = all_iterations
+
+        # Extract scaled and unscaled table
+        scaled_unscaled_match = re.findall(
+            r'Objective\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+).*?'
+            r'Dual infeasibility\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+).*?'
+            r'Constraint violation\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+).*?'
+            r'Complementarity\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+).*?'
+            r'Overall NLP error\.\.+:\s+([-+eE0-9.]+)\s+([-+eE0-9.]+)',
+            output,
+            re.DOTALL,
```

Comment:
Does the use of DOTALL and `.*?` make this a harder/more expensive regex?  Could we do the following:
```python
 scaled_unscaled_match = re.findall(
            r'Objective\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)\s*'
            r'Dual infeasibility\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)\s*'
            r'Constraint violation\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)\s*'
            r'Complementarity\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)\s*'
            r'Overall NLP error\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)',
            output,
)
```